src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/__tests__/ChartDataSetsConfiguration.test.tsx

```diff
@@ -978,5 +995,44 @@ describe('ChartDataSetsConfiguration', () => {
         },
       ]);
     });
+
+    test('removing all data sets', () => {
+      const handleChange = jest.fn();
+
+      render(
+        <ChartBuilderFormsContextProvider initialForms={testFormState}>
+          <ChartDataSetsConfiguration
+            meta={testSubjectMeta}
+            dataSets={[
+              {
+                filters: ['male'],
+                indicator: 'unauthorised-absence-sessions',
+              },
+              {
+                filters: ['male'],
+                indicator: 'unauthorised-absence-sessions',
+              },
+            ]}
+            onChange={handleChange}
+          />
+        </ChartBuilderFormsContextProvider>,
+      );
+
+      expect(screen.getAllByRole('row')).toHaveLength(3);
+
+      expect(handleChange).not.toHaveBeenCalled();
+
+      userEvent.click(screen.getByRole('button', { name: 'Remove all' }));
+
+      const modal = within(screen.getByRole('dialog'));
+      expect(modal.getByText('Remove all data sets')).toBeInTheDocument();
+      expect(
+        modal.getByText('Are you sure you want to remove all data sets?'),
+      ).toBeInTheDocument();
+
+      userEvent.click(modal.getByRole('button', { name: 'Confirm' }));
+
```

Comment:
We haven't actually check that the data sets have been removed

Reply:
I don't think it's possible in these tests as it's handled by ChartBuilder which then passes the datasets down to this component. I've added some tests for this for ChartBuilder, and a couple of other simple ones. I think most of the chart tabs functionality is tested in the individual tab tests but there may be more tests we want to add here in future.

Reply:
Ah good point! See that you've added some tests for `ChartBuilder` to test this though, nice one! :+1: